### PR TITLE
[16.0][MIG] base vat optional vies: Migration to 16.0

### PR DESCRIPTION
**Pull request by RodrigoBM**
#1472

### CODE REVIEW

**Comment by rafaelbn** [comment]
Hello @RodrigoBM , thank you!! 😄 

As a know issue in this module and commented recently by @alexis-via in Twitter https://twitter.com/alexisdlattre/status/1618377832090185730 we should improve of FIX this module in this points:

Premise: as de VAT number in the parent (commercial partner) is the same and should be the same for all chilldren , this field `vies_valid` too!!


So my proposal is simple:

- [ ] Propagate field to children: When VIES is valid and this module mark as true the boolean in the parent propagate it to all children contacts.
- [ ] Improve installation: when the module is installed, and there are already partners created, if we don't make anything the existing partner are not updated.
   - [ ] My proposal is create a wizard in the installation that ask to update all existing partners. If this is technically complicated or wrong for performance then create an action or something which objetive is update all existing partners.
   - [ ] Maybe the best way is create a installation script to be all recalculated correctly


This changes could be don't know if easy or later in a separate PR. I'm not blocking at all this great migration.

@moduon MT-2178

File: base_vat_optional_vies/models/res_partner.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        elif partner:
+            partner.update({"vies_passed": True})
+        return res
+
+    @api.constrains("vat", "country_id")
+    def check_vat(self):
+        self.update({"vies_passed": False})
+        for partner in self:
+            partner = partner.with_context(vat_partner=partner)
+            super(ResPartner, partner).check_vat()
```
Comment (Shide) [inline base_vat_optional_vies/models/res_partner.py]: Maybe can be improved to propagate vies_passed field to the children contacts:

```suggestion
		for partner in self.sorted(lambda p: bool(p.commercial_partner_id)):
			partner = partner.with_context(vat_partner=partner)
			if partner.commercial_partner_id:
				partner.update({"vies_passed": partner.commercial_partner_id.vies_passed})
			else:
				super(ResPartner, partner).check_vat()
```

**Review by Shide — APPROVED** [review]
Improvement that can be done in a separate PR, but not blocking migration

**Comment by RodrigoBM** [comment]
with this commit in base_vat https://github.com/odoo/odoo/commit/eff3b140cc88dd48b77948a577f195f2e1910fd8 this mig now fails. I am reviewing it

**Comment by RodrigoBM** [comment]
I have created pr for the test failure. https://github.com/OCA/account-financial-tools/pull/1577

**Comment by pedrobaeza** [comment]
Does this module now makes sense as Odoo already made the VIES check optional?

**Review by fcvalgar — APPROVED** [review]
LGTM.
Thank you @RodrigoBM

**Comment by fcvalgar** [comment]
> Hello @RodrigoBM , thank you!! smile
> 
> As a know issue in this module and commented recently by @alexis-via in Twitter https://twitter.com/alexisdlattre/status/1618377832090185730 we should improve of FIX this module in this points:
> 
> Premise: as de VAT number in the parent (commercial partner) is the same and should be the same for all chilldren , this field `vies_valid` too!!
> 
> So my proposal is simple:
> 
> * [ ]  Propagate field to children: When VIES is valid and this module mark as true the boolean in the parent propagate it to all children contacts.
> * [ ]  Improve installation: when the module is installed, and there are already partners created, if we don't make anything the existing partner are not updated.
>   
>   * [ ]  My proposal is create a wizard in the installation that ask to update all existing partners. If this is technically complicated or wrong for performance then create an action or something which objetive is update all existing partners.
>   * [ ]  Maybe the best way is create a installation script to be all recalculated correctly
> 
> This changes could be don't know if easy or later in a separate PR. I'm not blocking at all this great migration.
> 
> @moduon [Moduon - Task #2178](https://www.moduon.team/web#id=2178&model=project.task&view_type=form)

Initial validation of existing partners would be critical.

**Comment by fcvalgar** [comment]
> Does this module now makes sense as Odoo already made the VIES check optional?

It is a very user-friendly validation system without the need for external tools.

It also allows to trace, review and identify validation errors for non-technical users.

For me it is useful as a functional user.

**Comment by pedrobaeza** [comment]
Fran, but have you seen now how it's done on core? They have put a visible warning.

**Comment by fcvalgar** [comment]
> Fran, but have you seen now how it's done on core? They have put a visible warning.

It is probably my ignorance but it would only be a warning at the time of creation.

https://www.odoo.com/documentation/16.0/es/applications/finance/accounting/taxation/taxes/vat_validation.html

If I have a team member who wants to verify if that company has passed validation haven't the same information visually.

If this new validation would leave log. The financial and administrative departments  ❤️.

**Comment by pedrobaeza** [comment]
OK, I agree it can be useful to check it later, not only on VAT number onchange.

**Review by rafaelbn — APPROVED** [review]
Tested 👍 still needed Odoo core check it but don't allow to safe a VAT number si VIES is not valid. So the need is the same from v8 😄 

/ocabot merge patch

**Pull request merged**